scene/gui/text_edit.cpp

```diff
@@ -5112,6 +5094,18 @@ void TextEdit::select(int p_from_line, int p_from_column, int p_to_line, int p_t
 	queue_redraw();
 }
 
+void TextEdit::select_safe(int p_from_line, int p_from_column, int p_to_line, int p_to_column, int p_line, int p_column, int p_caret) {
```

Comment:
Not a fan of adding this method, would rather call `set_selection_mode` and `select` separately. And I wonder if some of these should be handled at a lower level like `set_line` and `insert_line_at` so this isn't needed in the first place.

Reply:
Hmm, but then how would the problem be solved? To reiterate, it happens because you can't provide new "pivots" for `select()`. What other ways is there to do this?

A) Add new method(s) for changing the "pivots" without calling `select()`?
B) Add optional arguments at the end of `select()` to preserve compat, which would be stupidly after `caret_idx`, or add them before `caret_idx`, which breaks compat and might be a bit bloated?
C) `set_selection_mode(get_selection_mode(), new_pivot_line, new_pivot_column)` ??

Or maybe something else?

Reply:
The something else would be making methods such as `insert_line_at` also adjust the "pivots". So then this "manual" adjustment won't be needed in the first place. But if not I'd rather go with C.

Reply:
I'd want to implement it the way you want, just not too sure I understand what you're suggesting. Mind taking this over to Rocketchat?